src/web-extension/manifest.json

```diff
@@ -33,7 +33,14 @@
   "browser_action": {
     "default_title": "Git Branch/Message",
     "default_icon": "icon-128.png",
-    "default_popup": "popup.html"
+    "default_popup": "popup.html",
+    "theme_icons": [
+      {
+        "light": "icon-light-128.png",
+        "dark": "icon-128.png",
+        "size": 128
+      }
```

Comment:
In theory yes, but I found it quite hard in practice, so I for not just left it as it was 😉. I read the guide on [choosing icon sizes](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_action#Choosing_icon_sizes) and it states:

> The browser action's icon may need to be displayed in different sizes in different contexts:
> * The icon is displayed by default in the browser toolbar, but the user can move it into the browser's menu panel (the panel that opens when the user clicks the "hamburger" icon). The icon in the toolbar is smaller than the icon in the menu panel.
> * On a high-density display like a Retina screen, icons needs to be twice as big.

☝️ this makes sense, but there's also this addition:

> If the browser can't find an icon of the right size in a given situation, it will pick the best match and scale it.

In my case it was always using the 16px one 😹 (looks hilarious - I first thought there wasn't an icon at all).

For the `default_icon` key, one could also use an SVG icon and the browser would scale it automatically to its needs. Unfortunately, the `theme_icons` don't support this though.

But I guess I'll have another look and see what's wrong with our configuration here 👀

Reply:
🤔 At least as per [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_action)  we should provide at least 16 and 32 px versions. I just thought that since it's an array anyway, we could just feed all of them to the browser and let it choose on its own. Or did you mean that when you do that it doesn't actually choose a good size?

Reply:
Yes, the latter. When I do it, it chooses the 16px version.

Reply:
🤦 Oh, alright. Then it's probably best to keep it like you did, with the largest version. 👍 

Thanks for the explanation. 💚

Reply:
Did some more reading and I think ™ I got it right now. See d56cd4df0099e5f64eb4f2204403fe4d194e4018 (details in commit message).

The icon is now sized correctly everywhere (also in Chrome - no color change there though).